examples/pool.rs

```diff
@@ -0,0 +1,67 @@
+//! examples/pool.rs
+
+#![deny(unsafe_code)]
+#![deny(warnings)]
+#![no_main]
+#![no_std]
+
+extern crate panic_semihosting;
+
+use cortex_m_semihosting::{debug, hprintln};
+use heapless::{
+    pool,
+    pool::singleton::{Box, Pool},
+};
+use lm3s6965::Interrupt;
+use rtfm::app;
+
+// Declare a pool of 128-byte memory blocks
+pool!(P: [u8; 128]);
+
+#[app(device = lm3s6965)]
+const APP: () = {
+    #[init]
+    fn init() {
+        static mut MEMORY: [u8; 512] = [0; 512];
+
+        // Increase the capacity of the memory pool by ~4
+        P::grow(MEMORY);
+
+        rtfm::pend(Interrupt::I2C0);
+    }
+
+    #[interrupt(priority = 2, spawn = [foo, bar])]
+    fn I2C0() {
+        // claim a memory block, leave it uninitialized and ..
+        let x = P::alloc().unwrap().freeze();
+
+        // .. send it to the `foo` task
+        spawn.foo(x).ok().unwrap();
```

Comment:
Same remark, why `.ok()`

Reply:
TL; DR because `[T; 128]` doesn't implement the `fmt::Debug` trait.

`unwrap` requires that the error type implements `Debug` but it doesn't in this case. The error type is the value the message that couldn't be sent; in this case the value has type `Box<A>` and that only implements when `A::Type: Debug`, which in this case it's `[u8; 128]` but `[T; N]` implement Debug only up to N = 32.